src/documents/IoCTestingFramework.md

```diff
@@ -443,6 +747,20 @@ fluid.tests.startRendering = function (asyncTest, instantiator) {
 };
 ```
 
-It's anticipated that such lengthy fixture lists will in practice themselves be generated from other more
-suitable and concise forms in JSON structures, either by Fluid's [Model Transformation](to-do/ModelTransformation.md) system, or otherwise -
-as well as making better use of a growing library of standardised assertion and triggering functions such as `fluid.tests.changeField`.
+Such repetitive sequences of standardised fixtures are best factored into reusable grades of type `fluid.test.sequenceElement` as seen
+in the [sequenceGrade example](#example-of-sequence-building-using-sequencegrade-) above. 
+
+## Design Discussion about the Testing Framework ###
+
+The framework was designed over October-December 2012, with initial call for implementation on the fluid-work mailing list at
+[October 31st](http://lists.idrc.ocad.ca/pipermail/fluid-work/2012-October/008615.html), continuing over a sequence of community meetings, and
+including a summary of work in progress on [December 5th](http://lists.idrc.ocad.ca/pipermail/fluid-work/2012-December/008652.html). The overall goals for the testing framework were presented as these:
+
+* To facilitate the testing of demands blocks that may be issued by integrators against components deployed in a particular (complex) context
+* To automate and regularise the work of "setup" and "teardown" in complex integration scenarios, by deferring this to our standard IoC infrastructure
+* To simplify the often tortuous logic required when using the "nested callback style" to test a particular sequence of asynchronous requests and responses (via events) issued against a component with complex behaviour
+* To facilitate the reuse of testing code by allowing test fixtures to be aggregated into what are the two standard forms for our delivery of implementation - a) pure JSON structures which can be freely interchanged and transformed, b) free functions with minimum dependence on context and lifecycle
+
+The framework was given a substantial spring-cleaning in October 2016, implementing significant new features such as promise-based fixtures, and priority-driven
+sequence grade assembly, and some support for compact invokers. A few significant bugs remain, especially when listing multiple "listener"-type fixtures
+adjacent in the sequence - see [FLUID-5502](https://issues.fluidproject.org/browse/FLUID-5502). The support for model change event fixtures is also very old-fashioned and needs to be reformed.
```

Comment:
Is there a JIRA filed for "The support for model change event fixtures is also very old-fashioned and needs to be reformed."?